service/src/main/java/gov/nasa/pds/api/registry/model/RequestAndResponseContext.java

```diff
@@ -357,7 +357,7 @@ public void setResponse(RestHighLevelClient client, SearchRequest request) throw
         long hitCount = hits.getTotalHits().value;
         if (hitCount == 1L) {
           this.formatters.get(this.format).setResponse(hits.getAt(0), this.fields);
-        } else {
+        } else if (hitCount > 1L) {
```

Comment:
Unless I'm missing something, the logic change here seems incorrect - the idea of that set of conditionals is "if the response yields a single result, as expected, throw an error, otherwise there is an incorrect number of responses (either zero, or multiple), in which case a descriptive error should be thrown".

If this change addresses a test failure, the test case should be adjusted instead imho.

Reply:
@alexdunnjpl it does not look like your commit introduced that error, since before, the `else` clause was already both referring to cases with 0 or >1 results. The change I am proposing it to exclude the case 0 result from the `else`.

I am not questioning the behavior since after my PR, we are getting the expected behavior, but there might be a better way, especially more readable way of doing that.

Reply:
@al-niessner could you chime in on whether the original logic (see 22a0970 pre-diff version) is a mistake, or correct given some context @tloubrieu-jpl and I are missing?